src/structured-source.ts

```diff
@@ -24,33 +24,31 @@
 
 import { upperBound } from 'boundary';
 
-export class Position {
-    constructor(line, column) {
-        this.line = line;
-        this.column = column;
-    }
+export type SourceRange = readonly [number, number];
+export type SourcePosition = {
+    readonly line: number,
+    readonly column: number,
 }
 
-export class SourceLocation {
-    constructor(start, end) {
-        this.start = start;
-        this.end = end;
-    }
+export type SourceLocation = {
+    readonly end: SourcePosition,
+    readonly start: SourcePosition,
 }
 
 /**
  * StructuredSource
- * @class
  */
-export default class StructuredSource {
+export class StructuredSource {
+    private readonly indice: number[];
```

Comment:
Make `indice` private.
Probably, this property is not useful. It is for testing.